src/AutoRest.CSharp/Common/Input/Source/CodeGenAttributes.cs

```diff
@@ -50,12 +50,13 @@ public bool TryGetCodeGenMemberAttributeValue(AttributeData attributeData, [Mayb
             return name != null;
         }
 
-        public bool TryGetCodeGenSerializationAttributeValue(AttributeData attributeData, [MaybeNullWhen(false)] out string propertyName, out IReadOnlyList<string>? serializationNames, out string? serializationHook, out string? deserializationHook)
+        public bool TryGetCodeGenSerializationAttributeValue(AttributeData attributeData, [MaybeNullWhen(false)] out string propertyName, out IReadOnlyList<string>? serializationNames, out string? serializationHook, out string? deserializationHook, out string? bicepSerializationHook)
```

Comment:
I think this would make sense if we can also use a dictionary for the attribute properties. Right now it is 1:1 with the attribute properties. I can file an issue to explore updating the attribute to use a dictionary.

Reply:
https://github.com/Azure/autorest.csharp/issues/4207